randovania/games/samus_returns/exporter/patch_data_factory.py

```diff
@@ -222,6 +225,47 @@ def _encode_hints(self) -> list[dict]:
             if isinstance(logbook_node, HintNode)
         ]
 
+        artifacts = [self.game.resource_database.get_item(f"Metroid DNA {i + 1}") for i in range(39)]
+        dna_hint_mapping = {}
+        hint_config = self.configuration.hints
+        if hint_config.artifacts != ItemHintMode.DISABLED:
+            dna_hint_mapping = guaranteed_item_hint.create_guaranteed_hints_for_resources(
+                self.description.all_patches,
+                self.players_config,
+                MSRHintNamer(self.description.all_patches, self.players_config),
+                hint_config.artifacts == ItemHintMode.HIDE_AREA,
+                artifacts,
+                False,
+            )
+        else:
+            dna_hint_mapping = {k: f"{k.long_name} is hidden somewhere on SR388." for k in artifacts}
+
+        # Shuffle DNA hints
+        hint_texts = list(dna_hint_mapping.values())
+        rng.shuffle(hint_texts)
+        dna_hint_mapping = dict(zip(artifacts, hint_texts))
+
+        dud_hint = "This Chozo Seal did not give any useful DNA hints."
+        actor_to_amount_map = [
+            ("s033_area3b", "LE_RandoDNA", 0, 6),
+            ("s050_area5", "LE_RandoDNA", 6, 12),
+            ("s065_area6b", "LE_RandoDNA", 12, 19),
+            ("s100_area10", "LE_RandoDNA_001", 19, 26),
+            ("s100_area10", "LE_RandoDNA_002", 26, 33),
+            ("s110_surfaceb", "LE_RandoDNA", 33, 39),
+        ]
+
+        for scenario, actor, start, end in actor_to_amount_map:
+            shuffled_hints = list(dna_hint_mapping.values())[start:end]
+            shuffled_hints = [hint for hint in shuffled_hints if "Hunter already started with" not in hint]
+            if not shuffled_hints:
+                shuffled_hints = [rng.choice(JOKE_HINTS + [dud_hint])]
+            hints.append(
+                {"accesspoint_actor": {"scenario": scenario, "actor": actor}, "text": "\n".join(shuffled_hints) + "\n"}
```

Comment:
Uh, I might be wrong here but I think this will throw an runtime error. You have mixed strings and tuples in `JOKE_HINTS` if the random selection is one of the tuple it should not work.

Reply:
I'll do more testing later

Reply:
Quickly checked it. Yeah, can throw an error like:
randovania\games\samus_returns\exporter\patch_data_factory.py", line 264, in _encode_hints
    {"accesspoint_actor": {"scenario": scenario, "actor": actor}, "text": "\n".join(shuffled_hints) + "\n"}
                                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: sequence item 0: expected str instance, tuple found